dask/array/random.py

```
@@ -20,6 +20,601 @@
+                if i not in lookup:
+                    arg.append(ar)
+                else:
+                    if isinstance(ar, Array):
+                        arg.append((lookup[i],) + block)
+                    else:  # np.ndarray
+                        arg.append((getitem, lookup[i], slc))
+            kwrg = {}
+            for k, ar in kwargs.items():
+                if k not in lookup:
+                    kwrg[k] = ar
+                else:
+                    if isinstance(ar, Array):
+                        kwrg[k] = (lookup[k],) + block
+                    else:  # np.ndarray
+                        kwrg[k] = (getitem, lookup[k], slc)
+            vals.append(
+                (_apply_random_func, self._generator, funcname, bitgen, size, arg, kwrg)
+            )
+
+        meta = _apply_random_func(
+            self._generator,
+            funcname,
+            bitgen,
+            (0,) * len(size),
+            small_args,
+            small_kwargs,
+        )
+
+        dsk.update(dict(zip(keys, vals)))
+
+        graph = HighLevelGraph.from_collections(name, dsk, dependencies=dependencies)
+        return Array(graph, name, chunks + extra_chunks, meta=meta)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def beta(self, a, b, size=None, chunks="auto", **kwargs):
+        return self._wrap("beta", a, b, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def binomial(self, n, p, size=None, chunks="auto", **kwargs):
+        return self._wrap("binomial", n, p, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def chisquare(self, df, size=None, chunks="auto", **kwargs):
+        return self._wrap("chisquare", df, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def choice(
+        self,
+        a,
+        size=None,
+        replace=True,
+        p=None,
+        axis=0,
```

Dask currently doesn't seem to support any other axis value other than 0 (because a must be one dimensional). We may want to raise an error proactively before the user calls `compute()`.

---

done with [2faf8ad](https://github.com/dask/dask/pull/9038/commits/2faf8ad7da9fe734309ccb98ea5c3e4c6a215b34) thank you